drivers/mtd/mtd_nvs.c

```diff
@@ -0,0 +1,2241 @@
+ * drivers/mtd/mtd_nvs.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * NVS: non volatile storage in flash
+ *
+ * Copyright (c) 2018 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sys/types.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <crc8.h>
+#include <debug.h>
+#include <errno.h>
+#include <sys/poll.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/mtd/mtd.h>
+#include <nuttx/mtd/configdata.h>
+#include <nuttx/mtd/nvs_priv.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define CONFIG_MTD_NVS_WRITE_BLOCK_SIZE 4
+#define NVS_WRITE_BLOCK_SIZE            CONFIG_MTD_NVS_WRITE_BLOCK_SIZE
+#define MIN(a, b)                       ((a) > (b) ? (b) : (a))
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/* MTD NVS opeation api */
+
+static int     mtdnvs_open(FAR struct file *filep);
+static int     mtdnvs_close(FAR struct file *filep);
+static ssize_t mtdnvs_read(FAR struct file *filep, FAR char *buffer,
+                           size_t buflen);
+static int     mtdnvs_ioctl(FAR struct file *filep, int cmd,
+                            unsigned long arg);
+static int     mtdnvs_poll(FAR struct file *filep, struct pollfd *fds,
+                           bool setup);
+
+/* Nvs external api, called in mtdnvs_ioctl() and mtdnvs_register() */
+
+/* Nvs internal api */
+
+/* Nvs flash operation api */
+
+/* Basic flash operation api */
```

Comment:
Why do we need this comments? I mean especially some that are not followed by any APIs